### PR TITLE
chore(deps): bump rsbuild to 0.4.3

### PR DESCRIPTION
**Pull request by 9aoy**
## Summary

bump rsbuild from 0.4.0 to 0.4.3 to fix devtools CI error, this is because https://github.com/web-infra-dev/rspack/pull/5568

image

https://github.com/web-infra-dev/modern.js/actions/runs/7946389959/job/21694067967

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: tests/e2e/builder/cases/lodash/index.swc.test.ts [hunk 1]
```diff
@@ -30,15 +30,15 @@ test('should optimize lodash bundle size when using SWC plugin', async () => {
   expect(size < 10).toBeTruthy();
 });
 
-test('should not optimize lodash bundle size when transformLodash is false and using SWC plugin', async () => {
+// TODO: fixed in rsbuild next version, see https://github.com/web-infra-dev/rsbuild/pull/1572
```
Comment (chenjiahan) [inline tests/e2e/builder/cases/lodash/index.swc.test.ts:33]: we can release Rsbuild 0.4.3 to fix this today.

Comment (9aoy) [reply tests/e2e/builder/cases/lodash/index.swc.test.ts:33]: ok, waiting for 0.4.3

Comment (chenjiahan) [reply tests/e2e/builder/cases/lodash/index.swc.test.ts:33]: released: https://github.com/web-infra-dev/rsbuild/releases/tag/v0.4.3

Comment (9aoy) [reply tests/e2e/builder/cases/lodash/index.swc.test.ts]: updated

**Review by Timeless0911 — APPROVED** [review]

**Pull request merged**